src/rgw/rgw_rados.cc

```diff
@@ -8354,8 +8354,12 @@ int RGWRados::delete_system_obj(rgw_obj& obj, RGWObjVersionTracker *objv_tracker
   return 0;
 }
 
-int RGWRados::delete_obj_index(rgw_obj& obj)
+int RGWRados::delete_obj_index(rgw_obj& _obj)
 {
+  rgw_obj obj = _obj;
+  if (obj.get_instance() == "null") {
```

Comment:
Yes, you are right about the null version. But here we just want to remove the obj index and I don't think we need to think about the versioning.  When we encode obj index whose instance is "null" , we set its instance empty before we send request to osd.